renderdoc/driver/vulkan/vk_debug.cpp

```diff
@@ -642,6 +642,10 @@ VulkanDebugManager::VulkanDebugManager(WrappedVulkan *driver)
 
   rm->SetInternalResource(GetResID(m_ArrayMSDescSet));
 
+  CREATE_OBJECT(m_PixelHistoryDescSet, m_ArrayMSDescriptorPool, m_ArrayMSDescSetLayout);
+
+  rm->SetInternalResource(GetResID(m_PixelHistoryDescSet));
```

Comment:
This resource shouldn't be created here if it's only needed on replay (and the same for the extra descriptor in the pool). The array<->MSAA stuff is initialised here because it's needed on capture as well.

Look at the members such as e.g. `struct PixelPicking` in `VulkanDebugManager` to see how they're stored and managed only in replay.